lib/actions/model.js

```diff
@@ -45,8 +46,14 @@ module.exports = {
       if (preSaveAction) dispatch(preSaveAction);
 
       const Model = storageManager.storage(brainstemKey).model;
-      const xhr = new Model({ id: modelId })
-        .save(attributes, saveOptions);
+      const model = new Model({ id: modelId });
+
+      if (!model.isValid()) {
+        dispatch(invalidAction(model.validationError));
+        return model.validationError;
+      }
+
+      const xhr = model.save(attributes, saveOptions);
```

Comment:
Definitely. We can cache the result of `.save()` which will return `false` if it's invalid so there's restructuring that can be done here.